engineyard/ey-cookbooks-stable-v5
# using passenger-config detach instead of kill -9. Ticket: CC-1184

## Description of your patch

Using passenger-config detach-process $PID instead of kill -9 $PID

## Recommended Release Notes

None.

## Estimated risk

Medium. Passenger processes can remain in memory.

## Components involved

Passenger 5

## Description of testing done

See QA instructions

## QA Instructions

1. Create simple Ruby application (https://github.com/engineyard/todo.git) can be used for this testing.
2. Boot the environment with the latest V5 stack version. Environment should be run with following parameters:
  * Application Server Stack - Passenger 5
  * Runtime - Ruby 2.3
  * Environment type - Single Instance
3. SSH to the instance.
4. Replace file `/etc/chef/recipes/cookbooks/passenger5/files/default/passenger_monitor` with the file `cookbooks/passenger5/files/default/passenger_monitor` provided in CC-1184 branch.
5. Run Chef with command `PATH=/usr/local/ey_resin/bin:$PATH /home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`
6. Ensure Chef run was finished without any errors.
7. Run the command `passenger-memory-stats | egrep -i "Passenger RubyApp:"` and save the output.
8. Run the command `/engineyard/bin/passenger_monitor <YOUR-APPLICATION-NAME> -l 1 -w 60`
9. Run the command `passenger-memory-stats | egrep -i "Passenger RubyApp:"` to ensure that Passenger processes have changed.
